helix-term/src/commands.rs

```diff
@@ -138,17 +138,57 @@ pub enum MappableCommand {
         name: &'static str,
         fun: fn(cx: &mut Context),
         doc: &'static str,
+        requirements: &'static [Req],
+        category: Category,
     },
 }
 
+#[derive(Copy, Clone)]
+/// Requirement to use command
+pub enum Req {
+    Lsp,
+    TreeSitter,
+    Dap,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Hash)]
+/// Category this command is related to
+pub enum Category {
+    None,
+    Movement,
+    Changes,
+    Shell,
+    SelectionManipulation,
+    Search,
+    MinorModes,
+}
+
+impl std::fmt::Display for Category {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(match self {
+            Category::None => "Other",
```

Comment:
Use `Category::Other`, `Unspecified` or something else, `None` is a very specific term in Rust IMO